File: native/android/VueNativeCore/src/main/kotlin/com/vuenative/core/Modules/LinkingModule.kt
package com.vuenative.core

import android.content.Context
import android.content.Intent
import android.net.Uri

class LinkingModule : NativeModule {
    override val moduleName = "Linking"
    private var context: Context? = null

    override fun initialize(context: Context, bridge: NativeBridge) {
        this.context = context.applicationContext
    }

    override fun invoke(method: String, args: List<Any?>, bridge: NativeBridge, callback: (Any?, String?) -> Unit) {
        val ctx = context ?: run { callback(null, "Not initialized"); return }
        when (method) {
            "openURL" -> {
                val url = args.getOrNull(0)?.toString() ?: run { callback(null, "Missing URL"); return }
                try {
                    val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url)).apply {
                        addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                    }
                    ctx.startActivity(intent)
                    callback(null, null)
                } catch (e: Exception) {
                    callback(null, e.message)
                }
            }
            "canOpenURL" -> {
                val url = args.getOrNull(0)?.toString() ?: run { callback(false, null); return }
                val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
                val canOpen = ctx.packageManager.resolveActivity(intent, 0) != null
                callback(canOpen, null)
            }
            "getInitialURL" -> callback(null, null)
            else -> callback(null, "Unknown method: $method")
        }
    }
}

File: native/ios/VueNativeCore/Sources/VueNativeCore/Modules/LinkingModule.swift
#if canImport(UIKit)
import UIKit

final class LinkingModule: NativeModule {
    var moduleName: String { "Linking" }

    func invoke(method: String, args: [Any], callback: @escaping (Any?, String?) -> Void) {
        switch method {
        case "openURL":
            guard let urlString = args.first as? String,
                  let url = URL(string: urlString) else {
                callback(nil, "Invalid URL")
                return
            }
            DispatchQueue.main.async {
                UIApplication.shared.open(url, options: [:]) { success in
                    callback(success, success ? nil : "Failed to open URL")
                }
            }
        case "canOpenURL":
            guard let urlString = args.first as? String,
                  let url = URL(string: urlString) else {
                callback(false, nil)
                return
            }
            DispatchQueue.main.async {
                callback(UIApplication.shared.canOpenURL(url), nil)
            }
        default:
            callback(nil, "Unknown method: \(method)")
        }
    }

    func invokeSync(method: String, args: [Any]) -> Any? { nil }
}
#endif
